sdk/webpubsub/azure-messaging-webpubsubclient/azure/messaging/webpubsubclient/models/_models.py

```diff
@@ -564,21 +949,9 @@ def clear(self) -> None:
             self.ack_map.clear()
 
 
-class StartStoppingClientError(Exception):
-    """Exception raised when the client is stopping but users want to start it again"""
-
-
-class StartNotStoppedClientError(Exception):
-    """Exception raised when the client is not stopped completely"""
-
-
-class StartClientError(Exception):
+class OpenClientError(AzureError):
     """Exception raised when fail to start the client"""
 
 
-class OpenWebSocketError(Exception):
-    """Exception raised when fail to open the websocket"""
-
-
-class DisconnectedError(Exception):
-    """Exception raised when the client is not connected to the service."""
+class WebPubSubConnectionError(AzureError):
+    """Exception raised when fail to open the websocket or connection is closed"""
```

Comment:
We shouldn't have two different exceptions for "failing to open", as we already have "OpenClientError". 
Is there a reason these are two different exceptions?

Reply:
Users need to open client then send message, which is 2 steps. "OpenClientError" may happen when open client, and "WebPubSubConnectionError" happen after open client when send message. So it is more clear with two different exceptions for users catch.

Reply:
hmm I'm still not clear on how a customer should handle the distinction here. From looking at the code - it seems to be that the `OpenClientError` _only_ happens in the case that the client is in the wrong state to be opened - is that correct?
Whereas a `WebPubSubConnectionError` indicates that something has failed/disconnected with the underlying websocket?
I don't think we should use a different error type to represent the same problem in two different places.

Reply:
I see. After looking through the code, I think we can simplify exceptions into two:
- OpenClientError （may happen during client open）
- SendMessageError (may happen after client open)

Reply:
Let me provide more detailed info about the scenarios:
- OpenClientError is quit clear and users just need to call open() again when it happens
- SendMessageError is more complicated:
a) error.error_detail is None, it means connection lost, and users can close it and open again
b) error.error_detail is not None,
   b.1) error_detail.name == "", it means no ack from service, users can send it again
   b.2) error_detail.name != "" , the error message is returned from service, and users shall handle it according to the                   message hint, which belongs business logic